storage/pipeline/input.go

```diff
@@ -521,80 +562,104 @@ func (m *Sealing) updateInput(ctx context.Context, sp abi.RegisteredSealProof) e
 	return nil
 }
 
-func (m *Sealing) calcTargetExpiration(ctx context.Context, ssize abi.SectorSize, cfg sealiface.Config) (minExpEpoch, targetEpoch abi.ChainEpoch, err error) {
-	var candidates []*pendingPiece
+// pendingPieceIndex is an index in the Sealing.pendingPieces map
+type pendingPieceIndex cid.Cid
 
-	for _, piece := range m.pendingPieces {
-		if piece.assigned {
-			continue // already assigned to a sector, skip
-		}
-		candidates = append(candidates, piece)
-	}
+type pieceBound struct {
```

Comment:
This type makes the assumption that:
* TermMin doesn't matter because DealEnd>=TermMin
  * So if we check that sector expiration is after DealEnd it's also after TermMin
* TermMax is always higher than DealEnd

Reply:
isnt term min and term max duration?

Reply:
They are, but as far as I understand, as soon as the deal is published, they get locked in to `dealStart+dealDuration` / `dealStart+termMax`, and this logic uses those absolute epochs